cli/azd/pkg/account/subscriptions.go

```diff
@@ -111,16 +113,32 @@ func (s *SubscriptionsService) ListSubscriptionLocations(
 	locations := []Location{}
 	pager := client.NewListLocationsPager(subscriptionId, nil)
 
+	includeLocation := func(location *armsubscriptions.Location) bool {
+		// By default, we only include physical regions that have a physical location. In practice this filters out the EUAP
```

Comment:
Just wondering... What are the implications if we always listed all locations regardless of it being physical?

Reply:
Good question - I wrote a small little program to answer this question:

https://gist.github.com/ellismg/5362960c75ef6cadb0093ce93962dd8d

It printed:

```
Non Physical Region centralusstage Central US (Stage) (Logical)
Non Physical Region eastusstage East US (Stage) (Logical)
Non Physical Region eastus2stage East US 2 (Stage) (Logical)
Non Physical Region northcentralusstage North Central US (Stage) (Logical)
Non Physical Region southcentralusstage South Central US (Stage) (Logical)
Non Physical Region westusstage West US (Stage) (Logical)
Non Physical Region westus2stage West US 2 (Stage) (Logical)
Non Physical Region asia Asia (Logical)
Non Physical Region asiapacific Asia Pacific (Logical)
Non Physical Region australia Australia (Logical)
Non Physical Region brazil Brazil (Logical)
Non Physical Region canada Canada (Logical)
Non Physical Region europe Europe (Logical)
Non Physical Region france France (Logical)
Non Physical Region germany Germany (Logical)
Non Physical Region global Global (Logical)
Non Physical Region india India (Logical)
Non Physical Region israel Israel (Logical)
Non Physical Region italy Italy (Logical)
Non Physical Region japan Japan (Logical)
Non Physical Region korea Korea (Logical)
Non Physical Region newzealand New Zealand (Logical)
Non Physical Region norway Norway (Logical)
Non Physical Region poland Poland (Logical)
Non Physical Region qatar Qatar (Logical)
Non Physical Region singapore Singapore (Logical)
Non Physical Region southafrica South Africa (Logical)
Non Physical Region sweden Sweden (Logical)
Non Physical Region switzerland Switzerland (Logical)
Non Physical Region uae United Arab Emirates (Logical)
Non Physical Region uk United Kingdom (Logical)
Non Physical Region unitedstates United States (Logical)
Non Physical Region unitedstateseuap United States EUAP (Logical)
Non Physical Region eastasiastage East Asia (Stage) (Logical)
Non Physical Region southeastasiastage Southeast Asia (Stage) (Logical)
Physical Region with No Location brazilus Brazil US (Physical)
Physical Region with No Location centraluseuap Central US EUAP (Physical)
Physical Region with No Location eastus2euap East US 2 EUAP (Physical)
```

There seem to be three "classes" of locations here:

1. The "Stage" regions, which I am pretty sure are not present in most subscriptions, since they feel like internal testing environments we happen to have access to since we are in the Microsoft tenant. (From experience, ARM doesn't seem to accept these regions)
2. A bunch of logical regions that correspond to geographies on the map (I think ARM rejects submitting to any of these)
3. A few which are physical which have no physical location, this includes the EUAP regions (which mostly work, however we have seen cases that certain types of resources can't be provisioned there, e.g. a log analytics workspace) and this "Brazil US" region, which I don't know anything about.

I suspect that at a minimum we almost always want to reject the logical regions.  I could imagine that maybe including the physical ones with no locations could be fine.

Reply:
If we can land a change that mirrors the Portal behavior, that would be the most ideal in my mind...

Reply:
The locations list in the portal depends on the subscription and service/resource you are creating.
There's a management api (IIRC) which allows you to list the supported regions for a resource. That would follow what the portal does